src/main/java/com/aws/greengrass/shadowmanager/PubSubIntegrator.java

```diff
@@ -0,0 +1,173 @@
+                case "update":
+                    UpdateThingShadowRequest request = new UpdateThingShadowRequest();
+                    request.setThingName(shadowRequest.getThingName());
+                    request.setShadowName(shadowRequest.getShadowName());
+                    request.setPayload(publishEvent.getPayload());
+
+                    this.updateThingShadowRequestHandler.handleRequest(request, SHADOW_MANAGER_NAME);
+                    logger.atTrace()
+                            .setEventType(LogEvents.UPDATE_THING_SHADOW.code())
+                            .kv(LOG_THING_NAME_KEY, shadowRequest.getThingName())
+                            .kv(LOG_SHADOW_NAME_KEY, shadowRequest.getShadowName())
+                            .log("Successfully updated shadow");
+                    break;
+                case "delete":
+                    DeleteThingShadowRequest deleteRequest = new DeleteThingShadowRequest();
+                    deleteRequest.setThingName(shadowRequest.getThingName());
+                    deleteRequest.setShadowName(shadowRequest.getShadowName());
+
+                    this.deleteThingShadowRequestHandler.handleRequest(deleteRequest, SHADOW_MANAGER_NAME);
+                    logger.atTrace()
+                            .setEventType(LogEvents.DELETE_THING_SHADOW.code())
+                            .kv(LOG_THING_NAME_KEY, shadowRequest.getThingName())
+                            .kv(LOG_SHADOW_NAME_KEY, shadowRequest.getShadowName())
+                            .log("Successfully deleted shadow");
+                    break;
+                case "get":
+                    GetThingShadowRequest getRequest = new GetThingShadowRequest();
+                    getRequest.setThingName(shadowRequest.getThingName());
+                    getRequest.setShadowName(shadowRequest.getShadowName());
+
+                    this.getThingShadowRequestHandler.handleRequest(getRequest, SHADOW_MANAGER_NAME);
+                    logger.atTrace()
+                            .setEventType(LogEvents.DELETE_THING_SHADOW.code())
+                            .kv(LOG_THING_NAME_KEY, shadowRequest.getThingName())
+                            .kv(LOG_SHADOW_NAME_KEY, shadowRequest.getShadowName())
+                            .log("Successfully retrieved shadow");
+                    break;
+                default:
+                    logger.atWarn()
+                            .kv(LOG_THING_NAME_KEY, shadowRequest.getThingName())
+                            .kv(LOG_SHADOW_NAME_KEY, shadowRequest.getShadowName())
+                            .kv(LOG_OPERATION, shadowRequest.getOperation())
+                            .log("Unable to perform shadow operation due to unknown value");
+                    break;
+            }
+            logger.atDebug().kv(LOG_TOPIC, topic)
+                    .log("Finished processing new shadow operation message over local PubSub");
+        } catch (IllegalArgumentException e) {
```

Comment:
can this be thrown from any where in the switch? If this is the only exception you are catching and it can only come from `extractShadowFromTopic` then put the exception handling nearer the source of the exception:
```
ShadowRequest shadowrequest = null;
try {
  shadowRequest = extractShadowFromTopic();
} catch (IllegalArgumentException e) {
  return;
}
switch () {
 ...
}
```